### ISSUE TITLE
Rust error parsing ignores spans with useful replacement text in children

### ISSUE THREAD

**Issue by tchajed** [flycheck/flycheck#1393]
Flycheck does not capture all of the information in Rust's error message for the following mistake (a misspelled macro):

```rust
fn main() {
  prnt!("")
}
```

Flycheck reports the following:

```text
     2   5 error           cannot find macro `prnt!` in this scope (rust)
     2   5 info            you could try the macro (rust)
```

but Rust's rendered error is:

```text
error: cannot find macro `printf!` in this scope
  --> src/main.rs:13:5
   |
13 |     printf!("%s\n", fname)
   |     ^^^^^^ help: you could try the macro: `print`
```

Here's Rust's JSON error (just the `message` field of this error):

```json
{
  "children": [
    {
      "children": [],
      "code": null,
      "level": "help",
      "message": "you could try the macro",
      "rendered": null,
      "spans": [
        {
          "byte_end": 355,
          "byte_start": 349,
          "column_end": 11,
          "column_start": 5,
          "expansion": null,
          "file_name": "src/main.rs",
          "is_primary": true,
          "label": null,
          "line_end": 13,
          "line_start": 13,
          "suggested_replacement": "print",
          "text": [
            {
              "highlight_end": 11,
              "highlight_start": 5,
              "text": "    printf!(\"%s\\n\", fname)"
            }
          ]
        }
      ]
    }
  ],
  "code": null,
  "level": "error",
  "message": "cannot find macro `printf!` in this scope",
  "rendered": "error: cannot find macro `printf!` in this scope\n  --> src/main.rs:13:5\n   |\n13 |     printf!(\"%s\\n\", fname)\n   |     ^^^^^^ help: you could try the macro: `print`\n\n",
  "spans": [
    {
      "byte_end": 355,
      "byte_start": 349,
      "column_end": 11,
      "column_start": 5,
      "expansion": null,
      "file_name": "src/main.rs",
      "is_primary": true,
      "label": null,
      "line_end": 13,
      "line_start": 13,
      "suggested_replacement": null,
      "text": [
        {
          "highlight_end": 11,
          "highlight_start": 5,
          "text": "    printf!(\"%s\\n\", fname)"
        }
      ]
    }
  ]
}
```

Note that "you could try the macro: `print`" is represented as "you could try the macro" plus a suggested replacement. It looks like [flycheck ignores spans in children under the assumption they aren't used](https://github.com/flycheck/flycheck/blob/31122714e1971d8403d9daf5815482bf5118c94d/flycheck.el#L5816-L5818), but this seems to be a counter example.

# Setup information

I'm using Rust nightly (1.25.0, 2018-02-01), but I double checked the JSON from Rust stable (1.23.0) and it has the same structure.

I didn't try to reproduce this in `emacs -Q`.

`M-x flycheck-verify-setup`:

```
Syntax checkers for buffer test.rs in rust-mode:

  rust-cargo (disabled)
    - may enable:  Automatically disabled!
    - predicate:   t
    - executable:  Found at /Users/tchajed/.cargo/bin/cargo
    - Cargo.toml:  Missing
    - Crate type:  lib
    - Binary name: Not required

  rust
    - may enable: yes
    - predicate:  t
    - executable: Found at /Users/tchajed/.cargo/bin/rustc

Flycheck Mode is enabled. Use SPC u C-c ! x to enable disabled checkers.

--------------------

Flycheck version: 32snapshot (package: 20180123.1419)
Emacs version:    25.2.1
System:           x86_64-apple-darwin13.4.0
Window system:    ns
```

operating system: OS X 10.11
emacs version: GNU Emacs 25.2.1 (x86_64-apple-darwin13.4.0, NS appkit-1265.21 Version 10.9.5 (Build 13F1911)) of 2017-04-21
Flycheck version: 32snapshot (package: 20180123.1419)

**Comment by fmdkdd** [comment]
Thank you for the report.  

I think I fixed that in #1385.  Could you try that branch (or just the patch to `flycheck-rustc-parse`) and see if that works for you?

**Comment by tchajed** [comment]
Sorry for the late response - I can confirm this is fixed on the master branch, and Spacemacs gets dependencies from MELPA rather than MELPA Stable so that version is used by default.
